UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -551,6 +559,17 @@ public void InitializeSensors()
             m_VectorSensorBuffer = new float[numFloatObservations];
         }
 
+        void InitializeRewardProvider()
+        {
+            // Look for a legacy reward provider.
+            var rewardProviderComponent = GetComponent<RewardProviderComponent>();
+            if (rewardProviderComponent == null)
+            {
+                rewardProviderComponent = gameObject.AddComponent<CumulativeRewardProviderComponent>();
```

Comment:
Why do we need to add the component ? Is it so we can see the curve on the Inspector? Are we keeping this in eventually?

Reply:
I imagined it like the sensor/sensorComponent style.  Where maybe in the future we'd come up with something that could be drag-and-drop for reward providers.  I think having the UI with the reward timeline is an improvement over the console printouts.  I'd like to keep it for now.